### PR TITLE
[internal][tracing] Move tracing to new directory

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR moves `promptflow.tracing` to separate src folder `src/promptflow-tracing`, and updates the `setup.py` to install new promptflow package `promptflow-tracing`: <https://pypi.org/project/promptflow-tracing/0.1.0b2/>

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-tracing/pyproject.toml [hunk 1]
```diff
@@ -36,17 +36,20 @@ packages = [
 
 [tool.poetry.dependencies]
 python = "<4.0,>=3.8"
+Jinja2 = "*"
 openai = "*"
```
Comment (crazygao) [inline src/promptflow-tracing/pyproject.toml:40]: openai>1.0

File: src/promptflow-tracing/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import json
+from pathlib import Path
+
+import pytest
+
+
+@pytest.fixture
+def dev_connections() -> dict:
+    with open(
+        file=(Path(__file__).parent.parent / "connections.json").resolve().absolute().as_posix(),
```
Comment (crazygao) [inline src/promptflow-tracing/tests/conftest.py:10]: connection needed?

File: src/promptflow-tracing/pyproject.toml (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow-tracing/pyproject.toml:40]: this dependency comes from openai injector, and we support all version, so no need to add boundary.

File: src/promptflow-tracing/tests/conftest.py (hunk 2 above)
Comment (zhengfeiwang) [reply src/promptflow-tracing/tests/conftest.py:10]: yes, we have test covering the openai injector logic, which requires connection - i use live here for now

File: src/promptflow-tracing/pyproject.toml [hunk 3]
```diff
@@ -36,17 +36,20 @@ packages = [
 
 [tool.poetry.dependencies]
 python = "<4.0,>=3.8"
+Jinja2 = "*"
```
Comment (wangchao1230) [inline src/promptflow-tracing/pyproject.toml:39]: why do we need jinja2 in promptflow-tracing?

File: src/promptflow-tracing/pyproject.toml (hunk 1 above)
Comment (wangchao1230) [inline src/promptflow-tracing/pyproject.toml:40]: add some comment on why we need these?

File: src/promptflow-tracing/pyproject.toml [hunk 4]
```diff
@@ -36,17 +36,20 @@ packages = [
 
 [tool.poetry.dependencies]
```
Comment (wangchao1230) [inline src/promptflow-tracing/pyproject.toml:37]: make promptflow depends on this package? and dependencies from there?

Comment (zhengfeiwang) [reply src/promptflow-tracing/pyproject.toml:37]: will add this once we get b2 released

File: src/promptflow-tracing/pyproject.toml (hunk 3 above)
Comment (zhengfeiwang) [reply src/promptflow-tracing/pyproject.toml:39]: as offline sync, Lina will help remove this dependency

File: src/promptflow-tracing/pyproject.toml (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow-tracing/pyproject.toml]: added

File: src/promptflow-tracing/pyproject.toml (hunk 3 above)
Comment (zhengfeiwang) [reply src/promptflow-tracing/pyproject.toml]: cherry-picked Lina's commit

File: src/promptflow-tracing/pyproject.toml (hunk 4 above)
Comment (wangchao1230) [reply src/promptflow-tracing/pyproject.toml:37]: Let's get this PR merged

Comment (zhengfeiwang) [reply src/promptflow-tracing/pyproject.toml:37]: made promptflow depende on this, let's wait for the ci

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**